mlir/lib/Dialect/AMX/Transforms/LegalizeForLLVMExport.cpp

```diff
@@ -55,21 +55,25 @@ Value getStride(ConversionPatternRewriter &rewriter,
                 const LLVMTypeConverter &typeConverter, MemRefType mType,
                 Value base, Location loc) {
   assert(mType.getRank() >= 2);
-  int64_t last = mType.getRank() - 1;
+  int64_t preLast = mType.getRank() - 2;
   Type llvmInt64Type = IntegerType::get(&typeConverter.getContext(), 64);
   unsigned width = mType.getElementType().getIntOrFloatBitWidth();
   assert(llvm::isPowerOf2_64(width) && width >= 8);
   unsigned bytes = width >> 3;
-  if (mType.isDynamicDim(last)) {
-    // Dynamic size needs code to compute the stride at runtime.
+  int64_t offset;
+  SmallVector<int64_t, 4> strides;
+  getStridesAndOffset(mType, strides, offset);
```

Comment:
If you're assuming above that `stride(last) == 1`, then `strides[preLast] == getDimSize[last]`, so maybe we don't need to call this function and just check if `isDynamicDim(preLast)` here, then keep `mType.getDimSize(last)` below.

Reply:
Stride 1 in the last dimension doesn't guarantee `strides[preLast] == getDimSize[last]` because strides can be specified separately, e.g. `memref<16x32xbf16, strided<[64, 1]>>` like in the added test. I encountered such memrefs in our TritonCPU lowerings.

Reply:
I see what you mean. `getStridesAndOffset` will get the actual strides, regardless if they're implicit or explicit. There's a lot of afine computations in that function, so we should avoid calling it too many times, and fuse the two utility functions as described above.